Ginger/GingerCore/ALM/Qtest/ExportToQtest.cs

```diff
@@ -0,0 +1,594 @@
+                
+                //set item fields
+                foreach (ExternalItemFieldBase field in testSetFields)
+                {
+                    if (field.ToUpdate || field.Mandatory)
+                    {
+                        if (string.IsNullOrEmpty(field.SelectedValue) == false && field.SelectedValue != "NA")
+                            testSet[field.ID] = field.SelectedValue;
+                        else
+                            try { testSet[field.ID] = "NA"; }
+                            catch { }
+                    }
+                }
+
+                //post the test set
+                testSet.Name = businessFlow.Name;
+
+                try
+                {
+                    testSet.Post();
+                }
+                catch (Exception ex)
+                {
+                    if (ex.Message.Contains("The Test Set already exists"))
+                    {
+                        result = "Cannot export " + GingerDicser.GetTermResValue(eTermResKey.BusinessFlow) + "- The Test Set already exists in the selected folder. ";
+                        Reporter.ToLog(eLogLevel.ERROR, result, ex);
+                        return false;
+                    }
+
+                    //Searching for the testset in case it was created in ALM although getting exception
+                    TestSetFactory TSetFact = mTDConn.TestSetFactory;
+                    TDFilter tsFilter = TSetFact.Filter;
+                    TestSetTreeManager treeM = (TestSetTreeManager)mTDConn.TestSetTreeManager;
+                    ISysTreeNode testSetParentFolder = (ISysTreeNode)treeM.get_NodeByPath(uploadPath);
+
+                    try
+                    {
+                        tsFilter["CY_FOLDER_ID"] = "" + testSetParentFolder.NodeID + "";
+                    }
+                    catch(Exception e)
+                    {
+                        tsFilter["CY_FOLDER_ID"] = "\"" + testSetParentFolder.Path.ToString() + "\"";
+                        Reporter.ToLog(eLogLevel.ERROR, $"Method - {MethodBase.GetCurrentMethod().Name}, Error - {e.Message}", e);
+                    }
+
+                    List TestsetList = TSetFact.NewList(tsFilter.Text);
+                    foreach(TestSet set in TestsetList)
+                    {
+                        if(set.Name == businessFlow.Name)
+                        {
+                            testSet = set;
+                            break;
+                        }
+                    }
+                }
+
+                businessFlow.ExternalID = testSet.ID.ToString();
+
+                //Add missing test cases
+                TSTestFactory testCasesF = testSet.TSTestFactory;
+                foreach (ActivitiesGroup ag in businessFlow.ActivitiesGroups)
+                {
+                    if (existingActivitiesGroups.Contains(ag) == false && string.IsNullOrEmpty(ag.ExternalID) == false && ImportFromQC.GetQCTest(ag.ExternalID) != null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)